src/core/messages.ts

```diff
@@ -260,6 +285,215 @@ export interface Messages extends EmitsDiscontinuities {
   get channel(): Ably.RealtimeChannel;
 }
 
+/**
+ * A listener for summary message reaction events.
+ * @param event The message reaction summary event that was received. Use it
+ *   with {@link Message.with} to keep an up-to-date reaction count.
+ */
+export type MessageReactionListener = (event: MessageReactionSummaryEvent) => void;
+
+/**
+ * A listener for individual message reaction events.
+ * @param event The message reaction event that was received.
+ */
+export type MessageRawReactionListener = (event: MessageReactionRawEvent) => void;
+
+/**
+ * Add, delete, and subscribe to message reactions.
+ */
+export interface MessagesReactions {
+  /**
+   * Add a message reactions
+   * @param message The message to react to.
+   * @param type The type of reaction reference.
+   * @param reaction The reaction to add.
+   * @param count The count of the reaction for types that support it, default 1.
+   * @returns A promise that resolves when the reaction is added.
+   */
+  add(message: { serial: string }, type: MessageReactionType, reaction: string, count?: number): Promise<void>;
```

Comment:
Should we make `type` part of the room options? I can't think realistically of many instances whereby anyone will switching adhoc or using two at once. Even if we keep the in-method version as an optional parameter on the end, I think having the default in the options is valuable.

Reply:
Generally speaking - as we've got quite a few arguments here, with a few optionals, might be worth making a `MessageReactionParams` type that wraps them all?

Reply:
> asymmetric since you'll have a default for sending but not for receiving. Or do you suggest we add a default for receiving as well somehow?

On the receiving end, we give everyone everything, right? I don't think that sort of asymmetry is too bad. I'm just wary if we're requiring someone to pass in a type every single time its just something else they have to do wherein 99% of use-cases, one type will be all they need.

> Since there's only one optional, count, I'd argue to keep it as-is. If we make type optional as well then I'd change it to add(message, reaction, { type?, count? })

Just thinking of what we do with `Messages` - having an object-type param now would set us up for a similar API in the future

Reply:
We had a call about this, decision is:

1. Add the a default `type` as room option, and make `type` parameter optional in `reactions.add()`.
2. Use an object as parameter to match what we do with `room.messages.send()` and other methods.
3. On the receiving end (so `Message` interface) so make no change: you still have `message.reactions.<type>[reaction]`

We said no to:
1. Adding 3 (so 6 with deletes included) add reaction methods, one set for each type to avoid configs/defaults.
2. Adding a default `message.reactions` (to have the default type) along with `message.reactionTypes` (like current one with access to all), because typing the `message.reactions` correctly requires to use either `any`, `unknown` or generics at least up to the Room level but potentially to ChatClient.